exporter/signalfxexporter/metricdata_to_signalfxv2.go

```diff
@@ -0,0 +1,403 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signalfxexporter
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"unicode"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf"
+	"go.uber.org/zap"
+)
+
+// Some fields on SignalFx protobuf are pointers, in order to reduce
+// allocations create the most used ones.
+var (
+	// SignalFx metric types used in the conversions.
+	sfxMetricTypeGauge             = sfxpb.MetricType_GAUGE
+	sfxMetricTypeCumulativeCounter = sfxpb.MetricType_CUMULATIVE_COUNTER
+
+	// Array used to map OpenCensus metric descriptor to SignaFx metric type.
+	ocDescriptorToMetricType = [8]*sfxpb.MetricType{
```

Comment:
Does the metric description have anything that is equivalent to a delta/distributed counter value, e.g. one that indicates the incremental value since the last time it was reported?  These are generally not recommended but there are some integrations that require them (e.g. statsd data sources like to use them a lot and it's easier to just send on the delta value to the backend rather than keep state in the agent/collector).

Reply:
No, this will have to be handled is by having the "metrics receiver" of such cases to handle the delta changes.